Source/WebKit/UIProcess/Extensions/Cocoa/WebExtensionControllerCocoa.mm

```diff
@@ -140,19 +140,31 @@ - (void)_didDeleteLocalStorage:(NSNotification *)notification
     auto uniqueIdentifiers = FileSystem::listDirectory(m_configuration->storageDirectory());
     for (auto& uniqueIdentifier : uniqueIdentifiers) {
         String displayName;
-        if (!WebExtensionContext::readDisplayNameFromState(stateFilePath(uniqueIdentifier), displayName))
+        if (!WebExtensionContext::readDisplayNameFromState(stateFilePath(uniqueIdentifier), displayName)) {
+            RELEASE_LOG_ERROR(Extensions, "Failed to read extension display name from State.plist for extension: %{private}@", (NSString *)uniqueIdentifier);
             continue;
+        }
 
         for (auto type : types) {
             auto *storage = sqliteStore(storageDirectory(uniqueIdentifier), type, this->extensionContext(uniqueIdentifier));
-            if (!storage)
+            if (!storage) {
+                RELEASE_LOG_ERROR(Extensions, "Failed to create sqlite store for extension: %{private}@", (NSString *)uniqueIdentifier);
+
+                Ref record = recordHolder->recordsMap.ensure(uniqueIdentifier, [&] {
+                    return WebExtensionDataRecord::create(displayName, uniqueIdentifier);
+                }).iterator->value;
```

Comment:
This should likely move out of the block, so you don't duplicate it from inside `calculateStorageSize`.

Reply:
there's a `continue` below so we wouldn't hit the `calculateStorageSize` block.

Reply:
Right, but it can still be shared code with the code on line 161-163. You should only need to do this in one place.